packages/opencensus-propagation-stackdriver/src/index.ts

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as v1API from './v1';
+
+export interface SpanContext {
+  /** Trace ID */
+  traceId: string;
+  /** Span ID */
+  spanId: string;
+  /** Options */
+  options?: number;
+}
+
+/**
+ * An transport and environment neutral API for getting request headers.
+ */
+export interface HeaderGetter {
+  getHeader(name: string): string|string[]|undefined;
+}
+
+/**
+ * A transport and environment neutral API for setting headers.
+ */
+export interface HeaderSetter {
+  setHeader(name: string, value: string): void;
+}
+
+
+export interface Propagation {
+  extract(getter: HeaderGetter): SpanContext|null;
```

Comment:
We were planning to add a Propagation interface to Core package in the next few weeks. What do you think about to add this interface to Core package? It should be at :
```
opencensus-core/src/trace/propagation/types.ts
```
or maybe in a new PR.

Reply:
Here's the list of dependencies that `core` has:

```
  "dependencies": {
    "continuation-local-storage": "^3.2.1",
    "debug": "^3.1.0",
    "intercept-stdout": "^0.1.2",
    "log-driver": "^1.2.7",
    "semver": "^5.5.0",
    "shimmer": "^1.2.0",
    "uuid": "^3.2.1"
  }
```

I think the `propagation` modules should have close to zero dependencies. In my specific use case, I want to use it in a logging context, and I don't specifically do not want to bring in `continuation-local-storage` or other similar functionality. The only things I need from `core` are two abstract types.

I think this discussion is suggesting to me that `core` is perhaps not "core-enough". /cc @kjin 

> in the next few weeks. 

I need this propagation module right now. If there is refactoring possible – I'm +1 but let's remove the code from when you're ready in a few weeks. For now this type can live here.

Reply:
> I think this discussion is suggesting to me that `core` is perhaps not "core-enough".
This makes sense to me. We should discuss the possibility of an interfaces-only module.

Reply:
> ..  If there is refactoring possible – I'm +1 but let's remove the code from when you're ready in a few weeks. For now this type can live here.

Ok

Do you prefer to open yourself the PR to include those definitions to ``opencensus-core/src/trace/propagation/types.ts`` or do you mind if we do so? Of course, we'll gonna keep you as the author of the commit.